src/type.ts

```diff
@@ -28,10 +41,10 @@ export const TYPE_FROM_SHORT_TYPE = {
 
 /**
  * Get full, lowercase type name for a given type.
- * @param  {String} type
- * @return {String} Full type name.
+ * @param  type
+ * @return Full type name.
  */
-export function getFullName(type: string): string {
-  return TYPE_FROM_SHORT_TYPE[type.toUpperCase()] || // short type is uppercase by default
-         type.toLowerCase();
+export function getFullName(type: Type): Type {
+  return TYPE_FROM_SHORT_TYPE[Type[type].toUpperCase()] || // short type is uppercase by default
```

Comment:
I admit it looks a bit ugly but we somehow have to make sure that if we get garbage in that we get the right thing out. I think it's not too bad.